src/main/java/com/powsybl/openloadflow/equations/EquationSystem.java

```diff
@@ -152,6 +152,14 @@ public Equation<V, E> removeEquation(int num, E type) {
         if (equation != null) {
             Pair<ElementType, Integer> element = Pair.of(type.getElementType(), num);
             equationsByElement.get(element).remove(equation);
+            if (equationTermsByElement != null) {
```

Comment:
This is not working on my case and I don't know exactly why. If I put 
````
equationTermsByElement = null;
indexAllTerms();
````

Reply:
it is supposed to be equivalent but faster. The way you did it reindex everything which would be very slow.

Reply:
looking at test case I did, does it seem ok?